Remove unnecessary call to containingBlock in `RenderLayerModelObject::styleDidChange`

Ahmad-S792 opened the pull request:
#### 57e31469bc48345cdb037cf78966a638bf09f43e
<pre>
Remove unnecessary call to containingBlock in `RenderLayerModelObject::styleDidChange`

<a href="https://bugs.webkit.org/show_bug.cgi?id=287628">https://bugs.webkit.org/show_bug.cgi?id=287628</a>
<a href="https://rdar.apple.com/144778634">rdar://144778634</a>

Reviewed by Alan Baradlay.

Inspired by: <a href="https://chromium.googlesource.com/chromium/blink/+/b30a38e00230f0a41e9db778709bb1ec16916583">https://chromium.googlesource.com/chromium/blink/+/b30a38e00230f0a41e9db778709bb1ec16916583</a>

The `containingBlock()` check was added back in 7750@main to protect `updateLayerPosition()`,
which is not present there anymore and this is just to set repaint status.
This patch is to remove redundant tree traversal and clean-up code.

* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::styleDidChange):

Canonical link: <a href="https://commits.webkit.org/290364@main">https://commits.webkit.org/290364@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e01041eacc8998f273d283f3b2bb5e48f256c73

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89757 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9286 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44643 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94750 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40533 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9673 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17563 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69135 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26759 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92758 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7429 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81453 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49495 "Found 3 new API test failures: /TestWTF:WTF_EnumTraits.EnumMinMax, /TestWTF:WTF_EnumTraits.EnumNameValid, /TestWTF:WTF_EnumTraits.EnumNameSignedValues (failure)") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/7151 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/35833 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39659 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/77498 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36861 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96577 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16939 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12441 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/78008 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/17195 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77267 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77322 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19094 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21759 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20344 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/10118 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16952 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/22276 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16693 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/20145 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18475 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderLayerModelObject.cpp:
```diff
@@ -158,7 +158,7 @@ void RenderLayerModelObject::styleDidChange(StyleDifference diff, const RenderSt
                 setChildNeedsLayout();
             createLayer();
-            if (parent() && !needsLayout() && containingBlock())
+            if (parent() && !needsLayout())
```
Well, the question is whether we expect a parent or a containing block here? Let me take a look.

alanbaradlay replied:
Both checks were added 20 years ago containingBlock() -> https://commits.webkit.org/7750@main followed by parent() -> https://commits.webkit.org/8988@main but they were introduced to "protect" `updateLayerPosition` which is not even there anymore.
Since at this point all we wanna do is set the repaint status on the layer, probably neither of this is needed and instead we could just go with the `needsLayout()` check....but let's just remove the redundant tree traversal for now.

Ahmad-S792 committed:
Remove unnecessary call to containingBlock in `RenderLayerModelObject::styleDidChange`

https://bugs.webkit.org/show_bug.cgi?id=287628
rdar://144778634

Reviewed by Alan Baradlay.

Inspired by: https://chromium.googlesource.com/chromium/blink/+/b30a38e00230f0a41e9db778709bb1ec16916583

The `containingBlock()` check was added back in 7750@main to protect `updateLayerPosition()`,
which is not present there anymore and this is just to set repaint status.
This patch is to remove redundant tree traversal and clean-up code.

* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::styleDidChange):

Canonical link: https://commits.webkit.org/290364@main
